projects/igniteui-angular/schematics/utils/dependency-handler.ts

```diff
@@ -59,16 +59,28 @@ function getTargetedProjectOptions(project: WorkspaceProject<ProjectType>, targe
     throw new SchematicsException(`Cannot determine the project's configuration for: ${target}`);
 }
 
-export function getConfigFile(project: WorkspaceProject<ProjectType>, option: string): string {
-    const buildOptions = getTargetedProjectOptions(project, 'build');
-    if (!buildOptions[option]) {
-        throw new SchematicsException(`Could not find the project ${option} file inside of the ` +
-            `workspace config (${project.sourceRoot})`);
-    }
+export function getConfigFile(project: WorkspaceProject<ProjectType>, option: string, configSection: string): string {
+    switch (configSection) {
```

Comment:
Does this have to be a `switch` statement? Currently, the only thing it's serving is providing an "enum" for what we _except_ to be present in the project options (we're not even checking for it). You can change it to the following (I'd even get rid of the **section** error)
```suggestion
    const options = getTargetedProjectOptions(project, configSection);
    if (!options) {
        throw new SchematicsException(`Could not find matching ${configSection} section` +
            `inside of the workspace config ${project.sourceRoot} `);
    }
    if (!options[option]) {
        throw new SchematicsException(`Could not find the project ${option} file inside of the ` +
            `workspace config ${project.sourceRoot}`);
    }
    return options[option];
```

Reply:
Yes, this remove a bit of the duplication, and configSection could be further improved somewhat in the future, containing an enum with the meaningful sections in angular.json